### PR TITLE
Add `docker_exit_code()` function

### LINKED ISSUE #253: Add a way to leverage exit_code in docker_compose_run commands
We run a lot of `docker_compose_run` function, but the output is not really nice:

> In utils.php line 194:
>                                                                                                                                                                          
>   The command "'docker' 'compose' '-p' 'app' '-f' '/home/dalexandre/Dev/MyApp/infrastructure/docker/docker-compose.yml' '-f' '/home/dalexandre/Dev/MixWith  
>   TheMasters/infrastructure/docker/docker-compose.worker.yml' '-f' '/home/dalexandre/Dev/MyApp/infrastructure/docker/docker-compose.builder.yml' '--env-fil  
>   e=/home/dalexandre/Dev/MyApp/application/.env' 'run' '--rm' '--no-deps' 'builder' '/bin/sh' '-c' 'exec bin/phpunit'" failed.                               
>                                                                                                                                                                          
> 
> qa:tests

We should find a way to improve this.

The default output should not be the full command but only my task: `bin/phpunit`.

I think castor exit_code do that. Maybe it could be a context option?

### PR DESCRIPTION
**Pull request by lyrixx**
fixed #253

### CODE REVIEW

File: .castor/utils.php [hunk 1]
```diff
@@ -101,6 +101,28 @@ function create_default_context(): Context
     return new Context($data, pty: 'dev' === $data['env']);
 }
 
+function docker_exit_code(
```
Comment (damienalexandre) [inline .castor/utils.php:104]: Maybe `docker_compose_run_exit_code` is a more appropriate name?

File: .castor/utils.php [hunk 2]
```diff
@@ -101,6 +101,28 @@ function create_default_context(): Context
...
+): int {
+    $c = ($c ?? get_context())->withAllowFailure();
+
+    $process = docker_compose_run(
+        runCommand: $runCommand,
+        c: $c,
+        service: $service,
+        noDeps: $noDeps,
+        workDir: $workDir,
+        withBuilder: $withBuilder,
```
Comment (damienalexandre) [inline .castor/utils.php:120]: Could we use something like func_get_args() ?

File: .castor/utils.php (hunk 1 above)
Comment (lyrixx) [reply .castor/utils.php:104]: this is **very** long :(

File: .castor/utils.php (hunk 2 above)
Comment (lyrixx) [reply .castor/utils.php:120]: I sticked to castor implementation, and I prefer. Explicit > Implicit (Second point of Zen of Python :D)

**Review by damienalexandre — APPROVED** [review]

**Review by pyrech — APPROVED** [review]

File: .castor/qa.php [hunk 3]
```diff
@@ -5,11 +5,13 @@
...
-function all(): void
+function all(): int
 {
     install();
-    cs();
-    phpstan();
+    $cs = cs();
+    $phpstan = phpstan();
+
+    return max($cs, $phpstan);
```
[the remark this answers is not in the document]
Comment (lyrixx) [reply .castor/qa.php:14]: Maybe 👍🏼 but it must be tested - I fear for the output

**Pull request merged**